packages/app/src/app/hooks/useScript.ts

```diff
@@ -0,0 +1,60 @@
+// Based on https://usehooks.com/useScript/
+import { useState, useEffect } from 'react';
+
+const cachedScripts = [];
+
+export const useScript = (src: string) => {
+  const [state, setState] = useState({
+    loaded: false,
+    error: false,
+  });
+
+  useEffect(() => {
+    if (cachedScripts.includes(src)) {
+      setState({
+        loaded: true,
+        error: false,
+      });
+    } else {
+      cachedScripts.push(src);
+
+      const script = document.createElement('script');
```

Comment:
This can  be simplified with the spread operator

```js
const script = {
  ...document.createElement('script'),
  async: true,
  src,
};
```

Reply:
I would have to test this. To my understanding the script tag needs to be present in the DOM before you can set a src attribute on it, which triggers the network request.

Reply:
Hmm yes, the more "conventional" way for me is to not spread anything that isn't a plain object.